src/api/app/helpers/webui/staging/workflow_helper.rb

```diff
@@ -43,6 +43,22 @@ def progress(staging_project)
     end
   end
 
+  def reviewers_gravatar(request)
+    missing_reviews = request[:missing_reviews]
+    image_tags = []
+    missing_reviews.each do |review|
+      case review[:review_type]
+      when 'by_group'
+        group = Group.find_by_title(review[:by])
```

Comment:
This is super wasteful. You run a find_by per review - and mostly cover like 3-4 groups with that. You better preload those in the controller / model.

Reply:
you could also do it here but just with one/two queries